predicators/spot_utils/spot_localization.py

```diff
@@ -55,6 +58,9 @@ def __init__(self, robot: Robot, upload_path: Path,
 
         # Initialize robot pose, which will be updated in localize().
         self._robot_pose = math_helpers.SE3Pose(0, 0, 0, math_helpers.Quat())
+        # Create flag to indicate whether map has been initialized for the
+        # first time or not.
+        self._map_initialized = False
```

Comment:
sorry, actually, can we just get rid of this flag altogether and do the `set_localization` here within the `__init__`? in addition to being clearer, this has the advantage of removing superfluous calls to `get_robot_state` etc. in the function below